src/test/regress/sql/single_shard_table_udfs.sql

```diff
@@ -0,0 +1,101 @@
+CREATE SCHEMA null_dist_key_udfs;
+SET search_path TO null_dist_key_udfs;
+
+SET citus.next_shard_id TO 1820000;
+SET citus.shard_count TO 32;
+SET citus.shard_replication_factor TO 1;
+-- test some other udf's with single shard tables
+CREATE TABLE null_dist_key_table(a int);
+SELECT create_distributed_table('null_dist_key_table', null, colocate_with=>'none', distribution_type=>null);
+
+SELECT truncate_local_data_after_distributing_table('null_dist_key_table');
+
+-- should work --
+-- insert some data & create an index for table size udf's
+INSERT INTO null_dist_key_table VALUES (1), (2), (3);
+CREATE INDEX null_dist_key_idx ON null_dist_key_table(a);
+
+SELECT citus_table_size('null_dist_key_table');
+SELECT citus_total_relation_size('null_dist_key_table');
+SELECT citus_relation_size('null_dist_key_table');
+SELECT * FROM pg_catalog.citus_shard_sizes() WHERE table_name LIKE '%null_dist_key_table%';
+
+BEGIN;
+  SELECT lock_relation_if_exists('null_dist_key_table', 'ACCESS SHARE');
+  SELECT count(*) FROM pg_locks where relation='null_dist_key_table'::regclass;
+COMMIT;
+
+SELECT partmethod, repmodel FROM pg_dist_partition WHERE logicalrelid = 'null_dist_key_table'::regclass;
+SELECT master_get_table_ddl_events('null_dist_key_table');
+
+SELECT column_to_column_name(logicalrelid, partkey)
+FROM pg_dist_partition WHERE logicalrelid = 'null_dist_key_table'::regclass;
+
+SELECT column_name_to_column('null_dist_key_table', 'a');
+
+SELECT master_update_shard_statistics(shardid)
+FROM (SELECT shardid FROM pg_dist_shard WHERE logicalrelid='null_dist_key_table'::regclass) as shardid;
+
+SELECT truncate_local_data_after_distributing_table('null_dist_key_table');
+
+-- should return a single element array that only includes its own shard id
+SELECT shardid=unnest(get_colocated_shard_array(shardid))
+FROM (SELECT shardid FROM pg_dist_shard WHERE logicalrelid='null_dist_key_table'::regclass) as shardid;
+
+BEGIN;
+  SELECT master_remove_partition_metadata('null_dist_key_table'::regclass::oid, 'null_dist_key_udfs', 'null_dist_key_table');
+
+  -- should print 0
+  select count(*) from pg_dist_partition where logicalrelid='null_dist_key_table'::regclass;
+ROLLBACK;
+
+-- should fail --
```

Comment:
We currently return the id of the only shard for the given single shard table. The second parameter does not matter. Whether it is given NULL, or the name of a random column from the table, or even a random string that is not a column.

```sql
SELECT get_shard_id_for_distribution_column('null_dist_key_table', null);
 get_shard_id_for_distribution_column
--------------------------------------
                              1820000
(1 row)

SELECT get_shard_id_for_distribution_column('null_dist_key_table', 'a');
 get_shard_id_for_distribution_column
--------------------------------------
                              1820000
(1 row)

SELECT get_shard_id_for_distribution_column('null_dist_key_table', 'nonexistingcolumn');
 get_shard_id_for_distribution_column
--------------------------------------
                              1820000
(1 row)

```

That's also what we do for reference tables. Again, the second parameter does not matter.
It's because of the code block below.

```sql
	if (!HasDistributionKey(relationId))
	{
		List *shardIntervalList = LoadShardIntervalList(relationId);
		if (shardIntervalList == NIL)
		{
			PG_RETURN_INT64(0);
		}

		shardInterval = (ShardInterval *) linitial(shardIntervalList);
	}
```

I think we should keep doing the same thing for both reference and single shard tables. Either we should error out for both, or return the id of the only shard, again for both.
The current approach makes more sense to me, but I'd also be ok with erroring out.

Reply:
Never mind, I realized it's not on the list of functions that this PR is supposed to test. @onurctirtir I'm missing these functions in your testing document.

- [ ] alter_old_partitions_set_access_method
- [ ] create_time_partitions
- [ ] distributed_tables_colocated
- [ ] drop_old_time_partitions
- [ ] fix_pre_citus10_partitioned_table_constraint_names
- [ ] get_colocated_table_array
- [ ] get_missing_time_partition_ranges
- [ ] get_shard_id_for_distribution_column
- [ ] master_remove_distributed_table_metadata_from_workers
- [ ] master_update_table_statistics
- [ ] relation_is_a_known_shard
- [ ] run_command_on_colocated_placements
- [ ] run_command_on_placements
- [ ] run_command_on_shards
- [ ] shard_name

Reply:
Regarding the behaviour I agree that we should be consistent with reference tables. Let's leave this as is for now.